### DISCUSSION TITLE
Pass `form` Object to MuiForms and it should load values in the form.

### DISCUSSION THREAD

**Discussion by ahmedsakri** [manojadams/metaforms-core#60] [General]
**Is your feature request related to a problem? Please describe.**
I want to pass some values coming from API to the MuiForm tag. I am not able to do so currently. In current scenario. I have to pass it in the value key inside meta object.

Note: this is different from default Values.

**Describe the solution you'd like**

Something like below will be great.

```
<MuiForms
    schema={formSchema}
    onSubmit={(formData) => {
        console.log(formData, "Form Submitted Data");
    }}
    onChange={(formData) => {
        // console.log(formData, "Form Data");
    }}
    form={
        type: 'sampleActivity'
        properties: {
            name: "default_name",
            retry_config: {
                retry: 0,
                delay: 0,
                backoff: 1,

            }
        }
    }
/>
```        

**Describe alternatives you've considered**
I tried searching for alternatives but could not find one.

**Additional context**

**Comment by ahmedsakri** [comment]
@manojadams any thoughts on above?

**Comment by manojadams** [comment]
@ahmedsakri Thank you for your suggestion. 

Can you check the following questions:
- Would like to know any problem you are facing with the current approach for passing input data?
- Does the schema seems complex to understand?

**Comment by ahmedsakri** [comment]
@manojadams Thanks for the quick reply, below are my observations:

1. Would like to know any problem you are facing with the current approach for passing input data?
Answer: So, what i am currently doing is i am saving the form Object that i recieve directly in the API. So, in the future, if user wants to open that particular entry of the form what i have is the schema and the form object. With current approach, i will need to write a parser on the fly for the same which will inject the values in the value key inside the meta object.

2. Does the schema seems complex to understand?
Answer: The schema is quite easy to understand, it's just of the above scenario. I wanted it to be handled. The other library does this functionalities too:

- https://rjsf-team.github.io/react-jsonschema-form/
- https://jsonforms.io/examples/basic/


If you have any alternate approach, I am open to hear and implement this. Thanks a ton!

**Comment by ahmedsakri** [comment]
Also, @manojadams If you can look into [this](https://github.com/manojadams/mui-forms/issues/26) ticket. It will be very much helpful for me, as i need it badly at this moment.

**Comment by manojadams** [comment]
Ok. Instead of passing a form object, we can pass a similar object (key, value pair) that we are getting on submit event.
Will this solve your problem?

**Comment by ahmedsakri** [comment]
> Ok. Instead of passing a form object, we can pass a similar object (key, value pair) that we are getting on submit event. Will this solve your problem?

Yes, this is exactly what i need!

**Comment by ahmedsakri** [comment]
@manojadams I would like to follow-up with respect to this.

**Comment by manojadams** [comment]
@ahmedsakri It is WIP. You can expect a fix by eod.

**Comment by ahmedsakri** [comment]
@manojadams Any luck of releasing this?

**Comment by manojadams** [comment]
@ahmedsakri It will take some more time unfortunately.
By when do you need this?

**Comment by ahmedsakri** [comment]
@manojadams actually it would be nice if it is targeted by tomorrow. Currently #54 is what i can look forward to and complete the development.

**Comment by manojadams** [comment]
I have pushed the changes to following branch but testing is pending.
https://github.com/manojadams/metaforms-core/tree/feature-54

You can expect by eod once I am back from office.

**Comment by ahmedsakri** [comment]
> I have pushed the changes to following branch but testing is pending. https://github.com/manojadams/metaforms-core/tree/feature-54
> 
> You can expect by eod once I am back from office.

Thanks @manojadams Looking forward to it!

**Comment by manojadams** [comment]
@ahmedsakri  This feature has now been added with following version:
@manojadams/metaforms-core@3.0.42

**Example usage:**

https://codesandbox.io/p/sandbox/mui-forms-load-initial-form-data-mnplzh?file=%2Fpackage.json%3A13%2C5

**Comment by manojadams** [comment]
**Sample form data**
```
import MuiForms from "mui-forms";
import React from "react";
import schema from "./schema.json";

function InitialForm() {
  return (
    <MuiForms
      data={{
        basic_details: {
          first_name: "Bulma",
        },
        more_details: {
          personal_details: {
            last_name: "Bulmamoti",
          },
        },
      }}
      schema={schema}
      onSubmit={(formData) => {
        console.log(formData);
      }}
    />
  );
}

export default InitialForm;
```

**Comment by ahmedsakri** [comment]
> **Sample form data**
> 
> ```
> import MuiForms from "mui-forms";
> import React from "react";
> import schema from "./schema.json";
> 
> function InitialForm() {
>   return (
>     <MuiForms
>       data={{
>         basic_details: {
>           first_name: "Bulma",
>         },
>         more_details: {
>           personal_details: {
>             last_name: "Bulmamoti",
>           },
>         },
>       }}
>       schema={schema}
>       onSubmit={(formData) => {
>         console.log(formData);
>       }}
>     />
>   );
> }
> 
> export default InitialForm;
> ```

Thanks @manojadams!
